zerver/lib/send_email.py

```diff
@@ -276,7 +276,7 @@ def initialize_connection(connection: Optional[BaseEmailBackend] = None) -> Base
     if not connection:
         connection = get_connection()
 
-    if connection.open():
+    if connection is not None and connection.open():
```

Comment:
I think this should be done as a `assert connection is not None` the line before this; I think `get_connection` is supposed to always return a valid connection or throw an exception.  

It's only `None` the first time, before we've created the connection, I think.  This is:

https://docs.djangoproject.com/en/3.2/topics/email/#django.core.mail.get_connection